rules/aip0133/request_required_fields.go

```diff
@@ -33,10 +33,10 @@ var requestRequiredFields = &lint.MessageRule{
 		resourceMsgName := getResourceMsgNameFromReq(m)
 
 		// Rule check: Establish that there are no unexpected fields.
-		allowedRequiredFields := map[string]*builder.FieldType{
-			"parent": nil, // AIP-133
-			fmt.Sprintf("%s_id", strings.ToLower(strcase.SnakeCase(resourceMsgName))): nil,
-		}
+		allowedRequiredFields := stringset.New(
+			"parent",
+			fmt.Sprintf("%s_id", strings.ToLower(strcase.SnakeCase(resourceMsgName))),
```

Comment:
I would use the split type name (e.g. the `{Type Name}` portion of `type: "{Service Name}/{Type Name}"`), not the message name. We have a few resources that do not match the message name, as odd as that is, and it is not a requirement that they match.

Maybe even future proof this by checking the `singular` field first, then falling back on name parsing?

Sorry I missed this in my first go-around 🙈

Reply:
overall agreed, I wasn't super happy with how the message name was retrieved
originally, I just was opting for minimal-possible-change vs rewriting the
world.

But since you brought it up, I'll see what I can do here. agreed that we should
get the resource name from the singular, then type.

Reply:
ah, so one issue with this is how AIP-134 recommends the field name being the
singular of the resource.

If we had a standard field (e.g. resource) so we *know* that the field
represents the resource, it's much easier to grab that field and grab the
singular type. 

Actually, I think I can do this by:

1. making this a methodrule, and matching on "Create".
2. grabbing the output type, which *must* be the resource.
3. using that annotation to get the intended prefix name.

Okay, I'll try that.

Reply:
> If we had a standard field (e.g. resource) so we know that the field
represents the resource, i

Yeah I mean you can look up the resource message field in the same request message, there must only be one, so you wouldn't have to make it a `MethodRule`, but I see how going via response type would make it more fool proof.

Make sure to use the `utils.GetResponseType` that handles LROs as well :)

Reply:
I might as well - I do think there's some more refactoring to be done here, but
I'll do that incrementally and after the cloud standards when I have a few more
cycles. 

Addressed! PTAL.